hpke.go

```diff
@@ -150,6 +150,11 @@ func (cp contextParameters) aeadKey() []byte {
 	return cp.suite.KDF.Expand(cp.secret, context, cp.suite.AEAD.KeySize())
 }
 
+func (cp contextParameters) exporterSecret() []byte {
```

Comment:
Is there a reason we have exporterSecret / key / nonce as methods, rather than just deriving them when we create the contextParameters?

Reply:
It seems to give us a clean separation between the `keySchedule` output and `cipherContext` structs.